src/psyclone/psyir/nodes/psy_data_node.py

```diff
@@ -0,0 +1,536 @@
+            # pylint: enable=too-many-boolean-expressions
+            # Valid PSyData names have been provided by the user.
+            self._module_name = name[0]
+            self._region_name = name[1]
+            self.set_region_identifier(self._module_name, self._region_name)
+
+    # -------------------------------------------------------------------------
+    @property
+    def region_identifier(self):
+        ''':returns" the unique region identifier, which is a tuple \
+            consisting of the module name and region name.
+        :rtype: 2-tuple (str, str)'''
+        return self._region_identifier
+
+    # -------------------------------------------------------------------------
+    def set_region_identifier(self, module_name, region_name):
+        '''Defines a unique region identifier based on module- and region-name
+        by simply concatenating these two strings with a "-". The region-name
+        is unique in the module, so concatenating the strings will result
+        in a unique region name.
+
+        :param str module_name: name of the module.
+        :param str region_name: name of the region.
+        '''
+        self._region_identifier = (module_name, region_name)
+
+    # -------------------------------------------------------------------------
+    def __str__(self):
+        ''' Returns a string representation of the subtree starting at
+        this node. '''
+        result = "{0}Start[var={1}]\n".format(self._text_name, self._var_name)
+        for child in self.psy_data_body.children:
+            result += str(child)+"\n"
+        return result+"{0}End[var={1}]".format(self._text_name, self._var_name)
+
+    # -------------------------------------------------------------------------
+    @property
+    def psy_data_body(self):
+        '''
+        :returns: the Schedule associated with this PSyData region.
+        :rtype: :py:class:`psyclone.psyir.nodes.Schedule`
+
+        :raises InternalError: if this PSyData node does not have a Schedule \
+                               as its one and only child.
+        '''
+        from psyclone.psyir.nodes import Schedule
+        if len(self.children) != 1 or not \
+           isinstance(self.children[0], Schedule):
+            raise InternalError(
+                "PSyData node malformed or incomplete. It should have a "
+                "single Schedule as a child but found: {0}"
+                .format([type(child).__name__ for child in self.children]))
+        return self.children[0]
+
+    # -------------------------------------------------------------------------
+    def _add_call(self, name, parent, arguments=None):
+        '''This function adds a call to the specified method of
+        self._var_name to the parent.
+
+        :param str name: name of the method to call.
+        :param parent: parent node into which to insert the calls.
+        :type parent: :py:class:`psyclone.psyir.nodes.Node`
```

Comment:
`CallGen` is in f2pygen which means that `parent` is not a `Node` but an instance of `BaseGen`.